### PR TITLE
[jobs] testing: reduce test flakiness

### LINKED ISSUE #2818: jobs.v3.api_client.location_search_sample_test: test_location_search_sample failed
jobs.v3.api_client.location_search_sample_test: test_location_search_sample failed
buildID: 2b3daa3ff1daa16903d67e8ea2d4e4e11b51ad0e
buildURL: https://source.cloud.google.com/results/invocations/6a943e7d-45ec-417e-9e99-0e8798d07271
status: failed

### LINKED ISSUE #2945: jobs.v3.api_client.auto_complete_sample_test: test_auto_complete_sample failed
jobs.v3.api_client.auto_complete_sample_test: test_auto_complete_sample failed
buildID: 0715f7d6ccaf22555493299f52bcbee92e176d08
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/91a4ccd6-7b7f-4a73-8b04-3acb3a2c5f26), [Sponge](http://sponge2/91a4ccd6-7b7f-4a73-8b04-3acb3a2c5f26)
status: failed

### PR DESCRIPTION
**Pull request by tmatsuo**
fixes #2945
fixes #2818

There were 8 tests which has `time.sleep(10)` for waiting data
propagation or whatever. I chose to use `@eventually_consistent.call`
instead of sleep in those tests.

Bonus point is now the tests are significantly faster because it doesn't
always wait for 10 seconds.

The downside is it will take a long time on failure. We may want to
specify the retry count on the eventualy consistent decorator once the
following issue is fixed:

https://github.com/GoogleCloudPlatform/python-repo-tools/issues/25

### CODE REVIEW

File: jobs/v3/api_client/requirements-test.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 pytest==5.3.2
 gcp-devrel-py-tools==0.0.15
+google-cloud-core
```
Comment (leahecole) [inline jobs/v3/api_client/requirements-test.txt:3]: How is google-cloud-core being used? I'm not seeing it anywhere. I could easily be missing something though

Comment (tmatsuo) [reply jobs/v3/api_client/requirements-test.txt:3]: It is used by gcp-devrel-py-tools, which has a broken dependency.
See: https://github.com/GoogleCloudPlatform/python-repo-tools/issues/24

I'll fix them with upcoming PR.

File: jobs/v3/api_client/commute_search_sample_test.py [hunk 2]
```diff
@@ -12,12 +12,25 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import pytest
+import re
 
-def test_commute_search_sample(capsys):
-    import commute_search_sample
-    import re
+from gcp_devrel.testing import eventually_consistent
```
Comment (gguuss) [inline jobs/v3/api_client/commute_search_sample_test.py:18]: I believe we're discouraging use of gcp_devrel and instead preferring flaky or retry. Is there an advantage to eventually_consistent?

Comment (tmatsuo) [reply jobs/v3/api_client/commute_search_sample_test.py:18]: `eventually_consistent` is a thin wrapper around retrying, specialized to eventually consistent stuff. I'm not sticking to use gcp_devrel, but the code is more readable and descriptive. I'm more than happy to maintain that library, so there's no worry of it being not maintained.

If you strongly feel we should use `retrying`, I can do that though.

**Comment by tmatsuo** [comment]
Thanks! When we decide to use `retrying`, I'll make a change, but for now I'm just submitting this.

**Pull request merged**